gcc/rust/resolve/rust-forever-stack.hxx

```diff
@@ -0,0 +1,227 @@
+// Copyright (C) 2020-2023 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
```

Comment:
Why do we have two headers? Not sure i like the .hxx extension. Would prefer a different name or combine the header.

Reply:
Tbh hxx for template only implementations is pretty standard. We've discussed them a bit in the past with @CohenArthur and agreed on not using them when there is a non templated implementation, no `.hxx` file should exist along a `.cc` file for a same given header file.

I believe that when handled correctly those might reduce compile time, do not quote me on that though as they're included anyway with the header ?.

Reply:
Let's talk about this tomorrow during the community call :)

Reply:
By chance, I've just found that @rsandifo-arm's commit 73b7582775254b764fd92ddb252a33dc15872c69 "Add rtl-ssa" did add `*.inl` files for what appears to be a similar purpose (from a quick look)?

<https://github.com/Rust-GCC/gccrs/blob/73b7582775254b764fd92ddb252a33dc15872c69/gcc/rtl-ssa/internals.inl#L1>
<https://github.com/Rust-GCC/gccrs/blob/73b7582775254b764fd92ddb252a33dc15872c69/gcc/rtl-ssa/is-a.inl#L1>
<https://github.com/Rust-GCC/gccrs/blob/73b7582775254b764fd92ddb252a33dc15872c69/gcc/rtl-ssa/member-fns.inl#L20-L26>

(I have no opinion on `*.inl` vs `*.hxx`.)